Multisig/UI/Settings/KeyPickerController/KeyPickerController.swift

```diff
@@ -0,0 +1,188 @@
+        switch indexPath.section {
+
+        case Section.address:
+            let address = addresses[indexPath.row]
+
+            switch indexPath.row {
+
+            case Index.default:
+                let cell = tableView.dequeueCell(DefafultKeyTableViewCell.self, for: indexPath)
+                cell.setHeader("Default")
+                cell.setLeft("#1")
+                cell.setAddress(address)
+                cell.setSelected(isSelected(indexPath))
+
+                switch listState {
+                case .collapsed:
+                    cell.setDetail(nil)
+                    cell.separatorInset = UIEdgeInsets(top: 0, left: .greatestFiniteMagnitude, bottom: 0, right: 0)
+                case .expanded:
+                    cell.setDetail("Derived keys are generated from your seed phrase. Select a key you would like to import.")
+                    cell.separatorInset = .zero
+                }
+                return cell
+
+            default:
+                let cell = tableView.dequeueCell(DerivedKeyTableViewCell.self, for: indexPath)
+                cell.setLeft("#\(indexPath.row + 1)")
+                cell.setAddress(address)
+                cell.setSelected(isSelected(indexPath))
+                return cell
+            }
+
+        case Section.showMore:
+            let cell = tableView.dequeueCell(ButtonTableViewCell.self)
+            cell.height = listState == .collapsed ? 44 : 76
+            let label = listState == .collapsed ?
+                "Show more derived keys" :
+                "Show more"
+            cell.setText(label) { [unowned self] in
+                self.showMore()
+            }
+            return cell
+
+        default:
+            preconditionFailure()
+        }
+    }
+
+    private func isSelected(_ indexPath: IndexPath) -> Bool {
+        viewModel.selectedIndex == indexPath.row
+    }
+
+    private func showMore() {
+        var updatedPaths: [IndexPath] = []
+        var inserted: [IndexPath] = []
+        var deleted: [IndexPath] = []
+
+        if listState == .collapsed {
+            listState = .expanded
+            updatedPaths = [
+                IndexPath(row: Index.default, section: Section.address),
+                IndexPath(row: Index.showMore, section: Section.showMore)
+            ]
+
+            // we already have 1 address as a default one
+            inserted = (1..<viewModel.pageSize).map { IndexPath(row: $0, section: 0) }
+        } else {
+            inserted = (addresses.count..<(addresses.count + viewModel.pageSize))
+                .map { IndexPath(row: $0, section: 0) }
+
+            viewModel.generateAddressesPage()
+
+            if !viewModel.canLoadMoreAddresses {
+                deleted = [IndexPath(row: Index.showMore, section: Section.showMore)]
+            }
+        }
+
+        tableView.beginUpdates()
+        do {
+            tableView.deleteRows(at: deleted, with: .automatic)
```

Comment:
why so advanced? :) You could just redraw whole table view

Reply:
because it has good animation :)